pymc3/distributions/transforms.py

```diff
@@ -45,19 +41,26 @@ class Transform:
     Attributes
     ----------
     name: str
+        The name of the transform.
+    param_extract_fn: callable
+        A callable that takes a `TensorVariable` representing a random
+        variable, and returns the parameters required by the transform.
     """
 
+    __slots__ = ("param_extract_fn",)
     name = ""
 
-    def forward(self, x):
+    def forward(self, rv_var: TensorVariable, rv_value: TensorVariable) -> TensorVariable:
         """Applies transformation forward to input variable `x`.
         When transform is used on some distribution `p`, it will transform the random variable `x` after sampling
         from `p`.
 
         Parameters
         ----------
-        x: tensor
-            Input tensor to be transformed.
+        rv_var
+            The random variable being transformed
+        rv_value
+            The parameters required for the transform.
```

Comment:
`rv_value` doesn't sound very intuitive for something that holds the transform parameters. (I was confused by this above already.)

How about `rv` and `transform_params`?
Or `rv` and `params`?

Reply:
We need to be clear about the `rv_var` and `rv_value[_var]` distinctions.

`rv_var`s are the "sample-space" variables that are produced by `RandomVariable` `Op`s.
`rv_value[_var]`s are the "measure-space" (or log-likelihood) variables that correspond to a specific value of an `rv_var`.

These are the same two types of variables described [here](https://github.com/pymc-devs/pymc3/pull/4489#discussion_r593541630), where the sloppy `P(X = x)` or `x ~ X` notation denotes the `rv_var` with `X` (i.e. the random variable), and the value variable with `rv_value[_var]`.

These transform methods are getting _those_ two variables, so any new name that involves "params" would be inaccurate, because the `rv_value` variable does not provide parameters.  The first argument, `rv_var`, _does_ provide access to a random variable's parameters via `rv_var.owner.inputs`, and&mdash;again&mdash;`rv_value` is a value that's compatible with the random variable `rv_var` (i.e. a value that could've been sampled from it).

Reply:
So `rv_var` is the tensor of the user-provided, observed values? (A `TensorConstant`?)

We might still want to copy parts of your explanation into the docstring.